### PR TITLE
Simplified the instantiation of Netty EventLoopGroup

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

In many places throughout the code base we are checking if the OS is Linux to enable the Netty Epoll based socket handling.

### Modifications

Grouped the instantation of the `EventLoopGroup` into one single util class.

### Result

Better organized code.

### CODE REVIEW

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * Return a SocketChannel class suitable for the given EventLoopGroup implementation
+     *
+     * @param eventLoopGroup
+     * @return
+     */
+    public static Class<? extends SocketChannel> getClientSocketChannelClass(EventLoopGroup eventLoopGroup) {
+        if (eventLoopGroup instanceof EpollEventLoopGroup) {
```
Comment (rdhabalia) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java]: can we still add check `Epoll.isAvailable()` to make sure os supports the group?

Comment (merlimat) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java]: The problem there is that in the ClientImpl we're also accepting an `EventLoopGroup` instance that can be created outside the PulsarClient. In that case, we don't know whether the particular `EventLoopGroup` was created with Epoll or not (even if it's supported), and the `SocketChannel` needs to match the `EventLoopGroup` implementation.

Comment (rdhabalia) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java]: Ok. actually, earlier we had similar check at [broker](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java?utf8=%E2%9C%93#L284).

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            return EpollServerSocketChannel.class;
+        } else {
+            return NioServerSocketChannel.class;
+        }
+    }
+
+    public static void enableTriggeredMode(ServerBootstrap bootstrap) {
+        if (Epoll.isAvailable()) {
+            bootstrap.childOption(EpollChannelOption.EPOLL_MODE, EpollMode.LEVEL_TRIGGERED);
```
Comment (coderzc) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java:75]: Why not use `ET` model, it seems to be more efficient

**Pull request merged**